dask/core.py

```diff
@@ -303,8 +304,23 @@ def subs(task, key, val):
             arg = subs(arg, key, val)
         elif type_arg is list:
             arg = [subs(x, key, val) for x in arg]
-        elif type_arg is type(key) and arg == key:
-            arg = val
+        elif type_arg is type(key):
+            # https://github.com/dask/dask/pull/2457
+            # mute a FutureWarning from NumPy that may bubble up here
+            # about "elementwise comparison failed, returning scalar instead"
+            # when doing something like Array(...).compute()
+            # since `arg` will happen to be a NumPy array
+            # In the future this will raise a
+            # ValueError: The truth value of an array with more than one ...
+            # We want to prepare for that, so we catch ValueError, but
+            # we also don't want users seeing the warning
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", FutureWarning)
+                try:
+                    if arg == key:
+                        arg = val
+                except ValueError:
+                    pass
```

Comment:
Does this affect performance?  `subs` gets called inside of tight inner loops.  Creating the context manager here might be enough to slow things down.

Reply:
I suspect that this does strongly impact performance.  We often end up removing function calls and context managers from tight inner loops.

My first though on this problem was to test both type and equality

    len(a) == len(b) and all(type(aa) == type(bb) for aa, bb in zip(a, b)) and all(aa == bb for aa, bb in zip(a, b))

I suspect that this would be faster, though I'm not sure.

Reply:
In a simple benchmark of

```python
import numpy as np
import dask.array as da

X = da.random.random((100, 2), (2, 2))
idx = np.array([0, 0, 1, 1])
X[idx].compute()
```

these are all about the same, but there's probably too much other stuff going on to measure `subs` accurately. I went with your suggestion.